### PR TITLE
adding json support

### PR DESCRIPTION
**Pull request by kfirbio**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ flake8 = "*"
 pytest-cov = "*"
 isort = "^5.10.1"
 mypy = ">=1"
-snowflake-connector-python = "^2.7.11"
+snowflake-connector-python = "^3.0.1"
```
Comment (kfirbio) [inline pyproject.toml:19]: was not sure about upgrading the connector, had some local issues with the old.
what do you think?

Comment (bentheiii) [reply pyproject.toml:19]: change it to `>=2.7.11`, to not break anything

File: tests/blackbox/test_syntax.py [hunk 2]
```diff
@@ -62,3 +62,14 @@ def test_sample_query(connection, queried_sample, expected_sample):
     connection.cursor().execute("insert into bar values (1, 'one'), (2, 'two'), (3, 'three'), (10, 'ten')")
     res = connection.cursor().execute(f"select x, y from bar sample ({queried_sample} rows);").fetchall()
     assert len(res) == expected_sample
+
+
+@mark.parametrize('query,expected', [
+    ('select x, y:a::varchar from bar;', [(1, '1'), (2, '2')]),
```
Comment (bentheiii) [inline tests/blackbox/test_syntax.py:68]: all the SF examples use string

File: yellowbox_snowglobe/snow_to_post.py [hunk 3]
```diff
@@ -106,6 +106,17 @@ class Rule:
...
+
+    # json query string
+    Rule(re.compile(r"(?ix)\b"
+                    r"(" + NAME_PATTERN + r"):(" + NAME_PATTERN + ")" + "::varchar"
+                    + r"\b"), replacement=r"\1->>'\2'"),
+
+    # json query int
+    Rule(re.compile(r"(?ix)\b"
+                    r"(" + NAME_PATTERN + r"):(" + NAME_PATTERN + ")" + "::number"
+                    + r"\b"), replacement=r"cast(\1->>'\2' as integer)"),
```
Comment (bentheiii) [inline yellowbox_snowglobe/snow_to_post.py:118]: if we don't support nested json lookups, add that to the known quirks

**Review by bentheiii — APPROVED** [review]

**Pull request merged**